File: backend/src/main/java/io/tingkai/money/entity/BankInfo.java
package io.tingkai.money.entity;

import java.util.UUID;

import io.tingkai.money.constant.DatabaseConstant;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Data
@Table(name = DatabaseConstant.TABLE_BANK_INFO)
public class BankInfo {
	@Id
	protected UUID id;
	protected String code;
	protected String name;
}

File: frontend/src/api/bankInfo.ts
import axios from 'axios';
import { BANK_INFO_GET_ALL_PATH } from './Constant';
import { ApiResponse } from '../util/Interface';

export interface BankInfo {
    code: string;
    name: string;
}

export interface BankInfoListResponse extends ApiResponse<BankInfo[]> { }

const getAll = async (): Promise<BankInfoListResponse> => {
    const response = await axios.get(BANK_INFO_GET_ALL_PATH);
    const data: BankInfoListResponse = response.data;
    return data;
};

export default { getAll };
